sequential/seq2pat.py

```diff
@@ -277,11 +277,17 @@ class Seq2Pat:
     sequences : List[list]
         A list of sequences each with a list of events.
         The event values can be all strings or all integers.
+    max_span_index: Union[int, None]
+        The value to apply a default maximum span constraint on items' indices, max_span_index=10 by default (10 items).
+        This is going to avoid regular users to run into a scaling issue when data contains long sequences but no
+        constraints are used to run the mining efficiently and practically. Power users can choose to drop this
+        constraint by setting it to be None or increase the maximum span of index as the system has resources to support.
     """
 
-    def __init__(self, sequences: List[list]):
-        # Validate input sequences
+    def __init__(self, sequences: List[list], max_span_index: Union[int, None] = 10):
```

Comment:
Not sure about the naming.. The "index" is about the Attribute but this parameter is about the "length" of the span. It is not index, really. 

So I would suggest `max_span` instead. 

On the contrary, `index_attr = Attribute()` in the code below makes total sense --because that thing is about the index

Reply:
`max_span` makes sense. I will change the naming.